### DISCUSSION TITLE
Using authenticated proxy in pylenium

### DISCUSSION THREAD

**Discussion by capytoste** [ElSnoMan/pyleniumio#229] [Q&A]
Hi, Carlos!
l've been struggling for a while trying to configure proxy settings in pylenium.json to test through an authenticated proxy at my workplace, and have finally run out of ideas on how to make it work.

my last attempts have been:

(1) trying to insert a "proxy" block within the capabilities section of pylenium.json:

```
"capabilities": {
   "proxy": {
    "httpProxy": "http://user:password@proxy_ip:proxy_port",
    "ftpProxy": "http://user:password@proxy_ip:proxy_port",
    "sslProxy": "http://user:password@proxy_ip:proxy_port",
    "proxyType": "MANUAL"
   }
},
```

(2) inserting a similar block at the beginning of the test:

```
def proxy_test(py):
    py.webdriver.capabilities.chrome['proxy'] = {
        "httpProxy": "http://user:password@proxy_ip:proxy_port,
        "ftpProxy": "http://user:password@proxy_ip:proxy_port,
        "sslProxy": "http://user:password@proxy_ip:proxy_port,
        "proxyType": "MANUAL",
    }

```
However py config process seems to pay no attention to any of them :-(

l would really appreciate if you can give me a hint on how to make it run.
Thanks in advance!

**Comment by ElSnoMan** [comment]
Hi @capytoste,

Thanks for reaching out! How do you do this with Selenium? I'm sure we can figure it out, but I haven't done this before 😅
